modules/standard/Regex.chpl

```diff
@@ -833,27 +837,29 @@ record regex {
   pragma "no doc"
   proc _search_match(text: exprType, anchor: c_int, param has_captures, ref captures):regexMatch
   {
+    /* This pattern is a bit ugly, but we'd like to avoid
+    unnecessarily copying `this` when local. And we need to maintain
+    the lifetime of the copied regex when remote. */
```

Comment:
I think it'd be clearer with "maintain the lifetime of the copied regex when remote" -> "deallocate the temporary copy when remote"

Reply:
Also, I think the way I can see to make this less ugly is to add `regex.localize()` similar to `string.localize()` that creates an aliased copy when it is local. (But of course that complicates the regex memory management). I don't think you need to do that in this PR but you could put a comment describing the alternative.

Reply:
Right, but saying something like "deallocate in the right place" vs the "lifetime" part would be nice. "lifetime" makes me think about a bunch of other issues (that certainly encompass deallocation).

Reply:
If "lifetime" encompasses "deallocate" when you read it, then I'm not inclined to change it

Reply:
For the `localize` wrapper, did we try something like:


```chpl
inline proc regex.localize() {
  var ret:regex(exprType);
  if _local || this.home == here {
    ret._regex = this._regex;
  } else {
    ret = this;
  }
  return ret;
}
```

Reply:
I think I considered that and the issue is that when `ret` gets `deinit` in the enclosing scope (like `_search_match`), it would do a refcount decrement (with no matching increment)

Reply:
Ah, yeah that's right. I think we could avoid that by adding a "refCount" or "isBorrowed" or something bool and set that in localize and avoid the refcount decrement if it's set. I'd be fine doing that as a followup cleanup though.

And actually I think string has to have similar functionality and it's probably called `isOwned` there so I'd just mimic that.

Reply:
Right. I think we should leave that for another PR though. Also arguably the string localize shouldn't return a string but rather a string view of sorts. That way we can have the lifetime checker handle the checking that the string view doesn't outlive the string.